packages/@atjson/document/test/serialize-test.ts

```diff
@@ -0,0 +1,1129 @@
+                end: 12,
+              }),
+              new Italic({
+                start: 0,
+                end: 12,
+              }),
+              new Bold({
+                start: 0,
+                end: 12,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCHello, world",
+        blocks: [
+          {
+            type: "paragraph",
+            attributes: {},
+          },
+        ],
+        marks: [
+          {
+            type: "bold",
+            range: "(1..13]",
+          },
+          {
+            type: "italic",
+            range: "(1..13]",
+          },
+        ],
+      });
+    });
+
+    test("objects", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "Missy Elliott’s\uFFFC“Supa Dupa Fly”",
+            annotations: [
+              new LineBreak({
+                start: 15,
+                end: 16,
+              }),
+              new ParseAnnotation({
+                start: 15,
+                end: 16,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCMissy Elliott’s\uFFFC“Supa Dupa Fly”",
+        blocks: [
+          {
+            type: "text",
+            parents: [],
+          },
+          {
+            type: "line-break",
+            selfClosing: true,
+            parents: ["text"],
+            attributes: {},
+          },
+        ],
+        marks: [],
+      });
+    });
+
+    test("sparse blocks", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "onetwothree",
+            annotations: [
+              new Paragraph({
+                start: 3,
+                end: 6,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCone\uFFFCtwo\uFFFCthree",
+        blocks: [
+          {
+            type: "text",
+            parents: [],
+          },
+          {
+            type: "paragraph",
+            parents: [],
+          },
+          {
+            type: "text",
+            parents: [],
+          },
+        ],
+        marks: [],
+      });
+    });
+
+    test("continuations in blocks", () => {
+      expect(
+        serialize(
+          new TestSource({
+            content: "onetwothreefour",
+            annotations: [
+              new Quote({
+                start: 0,
+                end: 11,
+              }),
+              new Paragraph({
+                start: 3,
+                end: 6,
+              }),
+            ],
+          })
+        )
+      ).toMatchObject({
+        text: "\uFFFCone\uFFFCtwo\uFFFCthree\uFFFCfour",
+        blocks: [
+          {
+            type: "quote",
+            parents: [],
+          },
+          {
+            type: "paragraph",
+            parents: ["quote"],
+          },
+          {
+            type: "text",
+            parents: ["quote"],
+          },
```

Comment:
This is an interesting case that shows an asymmetry in the text here on either side of the paragraph block. This `text` block is necessary only to close the previous `paragraph`, right? 

That does mean that if we ever want to query directly against the serialized format, looking for `text` blocks is going to be difficult since the text block is sometimes only implied. But perhaps we've already decided that we'll need a different format for querying?

Reply:
Correct, we are using this to short close the paragraph and continue the quote.

For querying purposes, you could look at the parents array or could use ranges that are provided using the `includeBlockRanges` option that I recommend that we use for querying purposes.